c++/src/kj/async-io-unix.c++

```diff
@@ -1101,6 +1101,17 @@ public:
         // Ignore disallowed address.
         return acceptImpl(authenticated);
       } else {
+        if (getPort() != 0) {
```

Comment:
Instead of assuming non-zero port implies TCP, it'd be more correct to extend `SocketAddress` with an accessor to read the address family, and then use `SocketAddress::getLocalAddress(fd).getFamily()`.

But even better might be to use `KJ_SYSCALL_HANDLE_ERRORS(::setsockopt(...))` and ignore EOPPNOTSUP. That way it's just one syscall instead of two...

Reply:
Yeah, that makes much more sense. I had originally been using the `PeerIdentity` returned in the `if` block below so that that no extra syscalls were needed to get the port, but then moved it up here to make sure it'd always run and didn't consider that doing so would add an extra syscall. I've switched to `KJ_SYSCALL_HANDLE_ERRORS`.